### PR TITLE
Tools: update binary file IO; Enable compression of extraction data directory.

### PR DESCRIPTION
**Pull request by gchen-apollo**

### CODE REVIEW

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py [hunk 1]
```diff
@@ -392,15 +392,18 @@ def save_msg_list_to_file(out_path, channel, msg_list):
     else:
         raise ValueError("saving func for {} not implemented".format(channel))
 
-def generate_compressed_file(extracted_data_path,
-                             compressed_file='sensor_data'):
+def generate_compressed_file(input_path, input_name,
+                            output_path, compressed_file='sensor_data'):
```
Comment (xiaoxq) [inline modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: Align with first argument.

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py [hunk 2]
```diff
@@ -466,24 +469,28 @@ def main():
...
 
-    ret = extract_data(args.record_path, output_path, channel_list,
+    ret = extract_data(args.record_path, output_abs_path, channel_list,
                        args.start_timestamp, args.end_timestamp, extraction_rate_dict)
     if ret is False:
         print('Failed to extract data!')
 
-    generate_compressed_file(output_path, args.compressed_file)
+    generate_compressed_file(input_path=args.output_path,
+                            input_name=output_relative_path,
```
Comment (xiaoxq) [inline modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: ditto

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py [hunk 3]
```diff
@@ -466,24 +469,28 @@ def main():
         sys.exit(1)
 
     # Create directory to save the extracted data
-    output_path = args.output_path + re.sub(r'[^0-9]', '', str(datetime.now()))
+    output_relative_path = re.sub(r'[^0-9]', '', str(datetime.now()))
```
Comment (xiaoxq) [inline modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: Better to call `strftime` to generate readable and program friendly string.

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py (hunk 1 above)
Comment (gchen-apollo) [reply modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: done

File: modules/tools/vehicle_calibration/sensor_calibration/extract_data.py (hunk 3 above)
Comment (gchen-apollo) [reply modules/tools/vehicle_calibration/sensor_calibration/extract_data.py]: done

**Pull request merged**